src/presenter/sort-presenter.js

```diff
@@ -0,0 +1,47 @@
+import SortView from '../view/sort-view.js';
+import {render, RenderPosition, replace, remove} from '../framework/render.js';
+import { SortType } from '../const.js';
+import { sort } from '../utils/sort.js';
+
+
+export default class SortPresenter{
+  #listContainer = null;
+  #currentSortType = SortType.DAY;
+  #onSortChange = null;
+  #sortView = null;
+
+  constructor({listContainer, onSortChange}){
+    this.#listContainer = listContainer;
+    this.#onSortChange = onSortChange;
+  }
+
+  #onSortTypeChange = (sortType) => {
```

Comment:
Перед установкой ещё можно проверять (по ТЗ) что он не такой же, чтобы не перерисовывать дважды при клике на один и тот же
```suggestion
  #onSortTypeChange = (sortType) => {
    if (this.#currentSortType !== sortType) {
      this.#currentSortType = sortType;
      this.#onSortChange?.();
    }
    }
```